### PR TITLE
Fix documentation to make clear the TURN is not replacing the STUN bu…

### PR DESCRIPTION
**Pull request by nickvergessen**
…t an addition

Signed-off-by: Joas Schilling <coding@schilljs.com>

### CODE REVIEW

**Comment by MichaIng** [comment]
@nickvergessen 
Is this true? I mean if TURN is used, peers to not need to know each others local IPs, making STUN obsolete? But maybe I also misunderstood things... Does TURN invoke STUN where required?

**Comment by nickvergessen** [comment]
So I just checked, and it seems that CoTURN has a STUN included, but since we don't know if everyone will use coturn, it's better to fix the statement that both might be required. Also claiming the STUN would not work is wrong ;)

**Comment by MichaIng** [comment]
@nickvergessen 
Indeed coTURN has an included STUN server as well, but since both are different protocols with different purpose, this does not necessarily mean that TURN itself includes STUN. Also the included STUN server can be disabled.

>  Also claiming the STUN would not work is wrong ;)

Okay indeed not optimal wording, but STUN will not result in a working P2P connection in mentioned cases while TURN allows to establish a relayed connection. So yeah STUN works, but it does not "work" to use it as connection solution.

And yes it makes sense to have STUN and TURN both configured, since the clients try to use P2P first, if required with STUN, and if this does not work TURN. If P2P with help of STUN already allows to establish a connection, TURN will be an unnecessary overhead.

What I want to make clear is that form what I understand, indeed TURN could be used as a complete replacement for STUN. STUN is a P2P connection helper and TURN connects two peers as a relay server. And since the clients connect to the TURN server, I guess that in this case STUN is indeed not used/invoked at all.

**Comment by jospoortvliet** [comment]
> What I want to make clear is that form what I understand, indeed TURN could be used as a complete replacement for STUN. STUN is a P2P connection helper and TURN connects two peers as a relay server. And since the clients connect to the TURN server, I guess that in this case STUN is indeed not used/invoked at all.

Ok, but when a connection can be made using 'just' STUN, is TURN used regardless? If it is, that would put unnecessary load on the TURN server...

**Comment by MichaIng** [comment]
@jospoortvliet
It is a question how the client deals it, but from what I remember from earlier discussions about Talk is that TURN is only used when P2P fails. So direct P2P is attempted, if that fails STUN is invoked, if that fails TURN relayed connection is used.

This also means that one should configure STUN in the Talk settings in every case, either leave the public one in place or user the own coTURN as STUN server as well.

File: docs/TURN.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ The configuration of Nextcloud Talk mainly depends on your desired usage:
 
 - Talk tries to establish a direct [peer-to-peer (P2P)](https://en.wikipedia.org/wiki/Peer-to-peer) connection, thus on connections **beyond the local network** (behind a [NAT](https://en.wikipedia.org/wiki/Network_address_translation) or router), clients do not only need to know each others public IP, but the participants local IPs as well. Processing this, is the job of a [STUN](https://en.wikipedia.org/wiki/STUN) server. As there is one preconfigured for Nextcloud Talk, still nothing else needs to be done.
 
-- But in many cases, e.g. **in combination with firewalls or [symmetric NAT](https://en.wikipedia.org/wiki/Network_address_translation#Symmetric_NAT)**, a STUN server will not work as well, and then a so called [TURN](https://en.wikipedia.org/wiki/Traversal_Using_Relays_around_NAT) server is required. Now no direct P2P connection is established, but all traffic is relayed through the TURN server, thus additional (at least internal) traffic and resources are used.
```
Comment (MichaIng) [inline docs/TURN.md:8]: Another suggestion:
> But in some cases, ..., a direct P2P connection is not possible, even with the help of a STUN server, and a so called TURN server needs to be configured additionally.

**Comment by MichaIng** [comment]
And another thing: https://help.nextcloud.com/t/howto-setup-nextcloud-talk-with-turn-server/30794/73?u=michaing
While `relay-ip` and `listening-ip` is often required for servers without NAT, `external-ip` can be required when the server is behind a NAT. Of course this cannot work if one has a dynamic IP (like me). Should I open a new PR to add this, or do we want to merge it here?

I think I'll remove the TURNS/TLS part completely (cleanup) and add the NAT/non-NAT party right below the main config block, so they are better visible.

**Comment by nickvergessen** [comment]
You mean as a follow up to https://github.com/nextcloud/spreed/pull/3221 ?
But lets make small steps, so try a new PR

File: docs/TURN.md (hunk 1 above)
Comment (jospoortvliet) [reply docs/TURN.md:8]: Good suggestion, that change is included now ;-)

**Comment by MichaIng** [comment]
@nickvergessen 
Ah great, I missed that PR. Hmm, from the speedbox docs I sounds like `external-ip` IP can only be required when the TURN server is behind a NAT: https://github.com/spreedbox/spreedbox/wiki/Use-TURN-server#run-turn-server-on-spreedbox-behind-nat
However it helped here in a case without NAT: https://help.nextcloud.com/t/howto-setup-nextcloud-talk-with-turn-server/30794/73?u=michaing
Not sure if `relay-ip` + `listening-ip` alone would have solved the issue as well, probably some more testing is required, or insights from coTURN devs, to be sure which of these is/can be needed in what scenario. I lack the possibility to test it since I can only do such on my home server behind a NAT with dynamic IP, where all kind of connections work well without any of these IP settings set 🤔.

**Pull request merged**